rust/plugin-lib/src/base_cache.rs

```diff
@@ -69,13 +69,13 @@ impl ChunkCache {
         }
 
         // We now know that the start of this line is contained in self.contents.
-        let mut start_off = self.cached_offset_of_line(line_num).unwrap();
+        let mut start_off = self.cached_offset_of_line(line_num).unwrap() - self.offset;
 
         // Now we make sure we also contain the end of the line, fetching more
         // of the document as necessary.
         loop {
             if let Some(end_off) = self.cached_offset_of_line(line_num + 1) {
```

Comment:
I think I'd prefer the `- self.offset` on both here, it feels asymmetrical to have different coordinates for `start_off` and `end_off`.

Reply:
it's on the next line? Do you mean you'd like it declared before use?

Reply:
I do see another bug here though, in comparing start_off to 0 in the next block

Reply:
Sorry if my comment was confusing. Just I'd like to start_off and end_off have the same units, either by subtracting off `self.offset` from both at the time they're bound, or neither, and have that applied at time of use. I see how doing the first is more than a little annoying because of the `if let`.